src/PackageProvider/InstalledRepositoryPackageProvider.php

```diff
@@ -8,7 +8,7 @@
 
 use function array_values;
 
-final class ComposerV2 implements PackageProviderDetectionInterface
+final class InstalledRepositoryPackageProvider implements PackageProviderDetectionInterface
```

Comment:
Probably yet another BC Break

Reply:
Yah, I still feel that we should not do SEMVER in composer plugins. This is not library code...

Maybe we should discuss this in the TSC? I really hate working on composer plugins - especially when it comes to BC compatibility.

Reply:
I prefer the simplest variant: one rule and for everything.
Anything else requires documentation and also complicates communication.

Reply:
Mostly worried about cases where consumers do rely on our symbols.

Documenting the limits/API of yhis library works: in that case, let's just make everything `@internal` abd we're good.